gitlab-ci-helper.sh

```diff
@@ -0,0 +1,180 @@
+#!/bin/bash
+set -euxo pipefail
+
+function build_book {
+  if [[ -z "$1" || -z "$2" || -z "$3" ]]; then
+    echo "Usage: build-book REF DIR INVALIDATION_PATH"
+    exit 1
+  fi
+
+  REF=$1
+  DIR=$2
+  INVALIDATION_PATH=$3
+
+  # Checkout the ref we were given
+  git checkout -q $REF
+  HEAD_REV=$(git rev-parse HEAD)
+
+  # Check if the existing ref matches
+  if [[ -f "$DIR" && -f "${DIR}/.rev" && "$(cat ${DIR}/.rev)" = "$HEAD_REV" ]]; then
+    echo "Cached book build for $REF found in $DIR!"
+    exit 0
+  fi
+
+  rm -rf $DIR
+  mkdir -p $DIR
+  echo "Building book for $REF..."
+  mdbook build book -d "../$DIR"
+
+  # Write the newly built rev to the rev file
+  echo "$HEAD_REV" >> $DIR/.rev
+
+  # Write the invalidation path since we just rebuilt
+  echo "$INVALIDATION_PATH" >> ./book-paths-updated
+}
+
+function build_docs_wasm {
+  if [[ -z "$1" || -z "$2" || -z "$3" ]]; then
+    echo "Usage: build-book REF DIR INVALIDATION_PATH"
+    exit 1
+  fi
+
+  REF=$1
+  DIR=$2
+  INVALIDATION_PATH=$3
```

Comment:
duplicated code
for info, variables are global (without scope) by default, so you can do all your parsing once